### DISCUSSION TITLE
DatePicker with french locale

### DISCUSSION THREAD

**Discussion by DDelbende** [Appsilon/shiny.fluent#136] [Q&A] [answered]
I need a DatePicker.shinyInput in french calendar ( "lundi" instead of "monday" for example).
Currently it renders english calendar names.
Even if I see in the documentation that it can be done,  I don't find my way out as there is no example.
**dateTimeFormatter ICalendarFormatDateCallbacks : Apply additional formating to dates, for example localized date formatting.** 

Could you help me with this :  DatePicker.shinyInput("toDate", value = Sys.Date(), label = "Fin")

**Comment by jakubsob** [comment, marked as answer]
Hi @DDelbende,

according to those two Github issues ([1](https://github.com/microsoft/fluentui/issues/2168), [2](https://github.com/microsoft/fluentui/issues/13802)) it's not possible to change strings in `DatePicker` by setting locale, but it can be done by providing strings that should be used by `DatePicker`. Please take a look at this example:
```r
library(shiny)
library(shiny.fluent)

ui <- fluentPage(
  DatePicker.shinyInput(
    "date",
    value = Sys.Date(),
    strings = list(
      months = list(
        "January",
        "February",
        "March",
        "April",
        "May",
        "June",
        "July",
        "August",
        "September",
        "October",
        "November",
        "December"
      ),
      shortMonths = list("Jan", "Feb", "Mar", "Apr", "May", "Jun", "Jul", "Aug", "Sep", "Oct", "Nov", "Dec"),
      days = list("Sunday", "Monday", "Tuesday", "Wednesday", "Thursday", "Friday", "Saturday"),
      shortDays = list("S", "M", "T", "W", "T", "F", "S"),
      goToToday = "Go to today",
      prevMonthAriaLabel= "Go to previous month",
      nextMonthAriaLabel = "Go to next month",
      prevYearAriaLabel = "Go to previous year",
      nextYearAriaLabel = "Go to next year"
    )
  )
)

server <- function(input, output) { }

shinyApp(ui, server)
```
You can substitute those strings with french translations.
Does this example help you solve your issue? If so please mark this comment as answer ;)

**Reply by DDelbende** [reply]
Great !

**Reply by kapitalistn** [reply]
Hi @jakubsob ,
This changed values in calendar, but the textbar remains unchanged. Are there any solutions?
P.S. Thanks for help!

![image](https://user-images.githubusercontent.com/84130032/198861675-209d3879-91aa-4a58-88ee-9a61f4e9de11.png)

**Reply by jakubsob** [reply]
Hi @kapitalistn,

the value in the bar remains unchanged, but it can be formatted using, e.g. `formatDate` field. According to the docs `formatDate` is:
> Optional method to format the chosen date to a string to display in the DatePicker 

To use it with this component, you'll need to pass a JS function that accepts a string and formats the date string according to your requirements. This trivial example below uses a function to always set the value in the bar to "date":
```r
library(shiny)
library(shiny.fluent)

ui <- fluentPage(
  DatePicker.shinyInput(
    "date",
    value = Sys.Date(),
    strings = list(
      months = list(
        "January",
        "February",
        "March",
        "April",
        "May",
        "June",
        "July",
        "August",
        "September",
        "October",
        "November",
        "December"
      ),
      shortMonths = list("Jan", "Feb", "Mar", "Apr", "May", "Jun", "Jul", "Aug", "Sep", "Oct", "Nov", "Dec"),
      days = list("Sunday", "Monday", "Tuesday", "Wednesday", "Thursday", "Friday", "Saturday"),
      shortDays = list("S", "M", "T", "W", "T", "F", "S"),
      goToToday = "Go to today",
      prevMonthAriaLabel= "Go to previous month",
      nextMonthAriaLabel = "Go to next month",
      prevYearAriaLabel = "Go to previous year",
      nextYearAriaLabel = "Go to next year"
    ),
    formatDate = JS("(date) => 'date'")
  )
)

server <- function(input, output) { }

shinyApp(ui, server)
```

**Comment by Patrikios** [comment]
Hallo together,
let's say I would define the `formatDate `fn like follows:
```JS
formatDate = JS("(date?: Date): string => 
        {
          if (!date) return '';
          const month = date.getMonth() + 1;  //  0  indexed
          const day = date.getDate();
          const year = date.getFullYear();
          return `${year}.${month}.${day}`;
        }")
```
However, there must be an issue because if I bind this funtion, nothing renders (the outputs on the page are canceled, they stay blank).
Am I havin the overall assumption icorrect?
